bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java

```diff
@@ -1154,6 +1159,73 @@ public static boolean format(ServerConfiguration conf,
         }
     }
 
+    /**
+     * Initializes bookie, by making sure that the journalDir, ledgerDirs and
+     * indexDirs are empty and there is no registered Bookie with this BookieId.
+     *
+     * @param conf
+     * @return
+     * @throws Exception
+     */
+    public static boolean initBookie(ServerConfiguration conf) throws Exception {
+        try (RegistrationManager rm = RegistrationManager.instantiateRegistrationManager(conf)) {
```

Comment:
RegistrationManager is only used after checking the directories. It would be good to defer creating a RM until it is needed, so it doesn't have to create zookeeper connection if there are non-empty directories exist.